backend/data_adapter.go

```diff
@@ -33,30 +33,58 @@ func (a *dataSDKAdapter) QueryData(ctx context.Context, req *pluginv2.QueryDataR
 			return RequestStatusFromError(innerErr), innerErr
 		}
 
+		if errors.Is(innerErr, context.Canceled) {
```

Comment:
Is it enough to just check context.Canceled, or should we also check grpcstatus.Code(err) == grpccodes.Canceled  ? Asking cause we are doing that in RequestStatusFromError.